sky/skylet/providers/azure/config.py

```diff
@@ -126,9 +130,27 @@ def _configure_resource_group(config):
         .properties.outputs
     )
 
+    # We should wait for the NSG to be created before opening any ports
+    # to avoid overriding the newly-added NSG rules.
+    nsg_id = outputs["nsg"]["value"]
+    nsg_name = nsg_id.split("/")[-1]
+    network_client = NetworkManagementClient(credentials, subscription_id)
+    backoff = common_utils.Backoff()
+    start_time = time.time()
+    while True:
+        nsg = network_client.network_security_groups.get(resource_group, nsg_name)
+        if nsg.provisioning_state == "Succeeded":
+            break
+        if time.time() - start_time > _WAIT_NSG_CREATION_NUM_TIMEOUT_SECONDS:
+            raise RuntimeError(
+                f"Fails to create NSG {nsg_name} in {resource_group} within "
+                f"{_WAIT_NSG_CREATION_NUM_TIMEOUT_SECONDS} seconds."
+            )
```

Comment:
Add a log here? “Waiting for …, sleeping for X seconds.”

Reply:
Added. Thanks!